src/lib/math/pcurves/pcurves_impl/pcurves_impl.h

```diff
@@ -698,14 +705,48 @@ class ProjectiveCurvePoint {
       }
 
       constexpr Self dbl_n(size_t n) const {
-         // TODO it is possible to optimize this by carrying over values from
-         // the previous iteration into the next
+         if constexpr(Self::A_is_minus_3) {
+            /*
+            Repeated doubling for A == -3
+            Cost: 2S + *2 + n*(4S + 4M + 2*2 + 1*3 + 4A)
 
-         Self pt = (*this);
-         for(size_t i = 0; i != n; ++i) {
-            pt = pt.dbl();
+            Naive doubling for A == -3 costs
+
+            n*(4S + 4M + 2*2 + 1*3 + 1*4 + 1*8 + 5A)
+
+            Algorithm 3.23 in
+            "Guide To Elliptic Curve Cryptography"
+            Hankerson, Menezes, Vanstone
+            */
+
+            auto nx = x();
+            auto ny = y();
+            auto nz = z();
+            ny = ny.mul2();
+            auto w = nz.square().square();
```

Comment:
Does it make sense to early-return on `n == 0`?

Reply:
Not really in practice - we call this with `n` derived from window sizes so `n` is always > 0